app/pages/reservation/PaymentFailed.js

```diff
@@ -7,12 +7,18 @@ import injectT from '../../i18n/injectT';
 
 function PaymentFailed({
   t,
+  resourceId
 }) {
   return (
     <div className="app-ReservationFailedPage__contentbox">
       <h2>{t('payment.title')}</h2>
       <p>{t('payment.text')}</p>
-      <Link className="app-ReservationFailedPage__link" to="/my-reservations">
+      {resourceId && (
+        <Link className="app-ReservationFailedPage__link" to={`/resources/${resourceId}`}>
```

Comment:
Should `history.goBack()` is better (so all queries was preserved)

Reply:
I thought about it but, the previous page is the Bambora form, and using `goBack()` prevents you from opening this URL in a new tab.

Reply:
Hah, true!